jwql/pull_jwql_branch.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+function echo_format {
+    echo "WARNING! the optional parameters should only be used during a JWQL release in production"
+    echo "Usage: $0 <branch> [-r|--reset_service] [-n|--notify <email@stsci.edu>]"
+}
+
+# Check if the required number of arguments are provided
+if [ "$#" -lt 1 ]; then
+    echo_format
+    exit 1
+fi
+
+# Set default values for optional flags
+reset=false
+notify=false
+recipient=""
+
+# Retrieve the branch_name from the command line argument
+branch_name=$1
+# Parse optional flags
+while [[ $# -gt 1 ]]; do
+    case "$2" in
+        -r|--reset_service)
+            reset=true
+            ;;
+        -n|--notify)
+            notify=true
+            recipient="$3"
+            shift
+            ;;
+        *)
+            echo "Error: Invalid option $2"
+            echo_format
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ "$notify" = true ]  && [ -z "$recipient" ]; then
+    echo_format
+    exit 1
+fi
+
+echo "Branch: $branch_name";
+echo "Reset: $reset";
+echo "Notify: $notify $recipient";
+
+# 1. Pull updated code from GitHub deployment branch (keep second checkout in case its already defined for some weird reason)
+git checkout -b $branch_name --track origin/$branch_name
+git checkout $branch_name
+git fetch origin $branch_name
+git pull origin $branch_name
+git fetch origin --tags
+
+# 2. Bring the server down
+if [ "$reset" = true ]; then
+    sudo /bin/systemctl stop jwql.service
+fi
+
+# 3. Install jwql
+pip install -e ..
```

Comment:
Looks like running the script is location dependent ie we have to be in the same directory as the script.